src/main/java/org/folio/service/manager/export/ExportResult.java

```diff
@@ -1,5 +1,86 @@
 package org.folio.service.manager.export;
 
-public enum ExportResult {
-  COMPLETED, IN_PROGRESS, ERROR;
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonObject;
+import org.folio.util.ErrorCode;
+
+@DataObject(generateConverter = true)
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+public class ExportResult {
+  private static final ExportResult IN_PROGRESS = new ExportResult(ExportStatus.IN_PROGRESS);
+  private static final ExportResult COMPLETED = new ExportResult(ExportStatus.COMPLETED);
+
+  private ExportStatus exportStatus;
+  private ErrorCode errorCode;
+
+  private ExportResult(ExportStatus status) {
+    this.exportStatus = status;
+  }
+
+  private ExportResult(ExportStatus status, ErrorCode errorCode) {
+    this.exportStatus = status;
+    this.errorCode = errorCode;
+  }
+
+  public ExportResult(JsonObject jsonObject) {
+    ExportResultConverter.fromJson(jsonObject, this);
+  }
+
+  public static ExportResult completed() {
+    return COMPLETED;
+  }
+
+  public static ExportResult inProgress() {
```

Comment:
Can this class be made enum, and leave the methods as is? Not sure how the dataobject works, but just checking

Reply:
I am not sure, but i think this is more informative solution, than simple enum.
As it is impossible in vert.x to make this class serializable on an event bus (it is possible for enum or json format, for example), so this annotation helps us to introduce ExportResult object in json format, and then remap it again to the object.

Reply:
This class can not be an enum, because enum can not hold the state. Enum fields are final static, but we need to hold the state of error, to pass error code back to the InputDataManager